msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -66,66 +107,273 @@ private void initializeSharedPreferenceFileManager(@NonNull final Context contex
+
+        if(didCurrentAccountChange(newAccount)){
+            callback.onAccountChanged(localAccount, newAccount);
+            return;
+        }
+
+        persistCurrentAccount(newAccountRecords);
+        callback.onAccountLoaded(newAccount);
+    }
+
+
+    @Override
+    public void signIn(@NonNull Activity activity,
+
+                @NonNull String[] scopes,
+                @NonNull AuthenticationCallback callback) {
+        acquireToken(
+                activity,
+                new String[]{"user.read"},
+                null, // account
+                null, // uiBehavior
+                null, // extraQueryParams
+                scopes, // extraScopes
+                null, // authority
+                callback,
+                null, // loginHint
+                null // claimsRequest
+        );
+    }
+
+    @Override
+    protected ILocalAuthenticationCallback getLocalAuthenticationCallback(final AuthenticationCallback authenticationCallback) {
+
+        return new ILocalAuthenticationCallback() {
+
+            @Override
+            public void onSuccess(ILocalAuthenticationResult localAuthenticationResult) {
+
+                //Get Local Authentication Result then check if the current account is set or not
+                MultiTenantAccount newAccount =  getAccountFromICacheRecordList(localAuthenticationResult.getCacheRecordWithTenantProfileData());
+
+                if(didCurrentAccountChange(newAccount)){
+                    //Throw on Error with UserMismatchException
+                    authenticationCallback.onError(new MsalClientException(MsalClientException.CURRENT_ACCOUNT_MISMATCH));
+                    return;
+                }else{
+                    persistCurrentAccount(localAuthenticationResult.getCacheRecordWithTenantProfileData());
+                }
+
+                IAuthenticationResult authenticationResult = AuthenticationResultAdapter.adapt(localAuthenticationResult);
+                authenticationCallback.onSuccess(authenticationResult);
+            }
+
+            @Override
+            public void onError(BaseException exception) {
+                MsalException msalException = MsalExceptionAdapter.msalExceptionFromBaseException(exception);
+                authenticationCallback.onError(msalException);
+            }
+
+            @Override
+            public void onCancel() {
+                authenticationCallback.onCancel();
+            }
+        };
+    }
+
+    private boolean didCurrentAccountChange(final @Nullable MultiTenantAccount newAccount){
+        MultiTenantAccount persistedAccount = getPersistedCurrentAccount();
+
+        if(persistedAccount == null){
+            if(newAccount == null){
+                return false;
+            }else{
+                return true;
+            }
+        }else{
+            if(persistedAccount.getId().equals(newAccount.getId())){
```

Comment:
nit: is Id case-sensitive?
If not, I'd prefer this to be equalIgnoreCase.

Reply:
Good question... I'll ignore case... these are guids typically and I don't think they are case sensistive
Will fix.